tests/testthat/test-utils.R

```diff
@@ -106,123 +106,134 @@ test_that("can convert and check R's dates", {
+    check_rel_param(-1, "rel_param"),
+    "All values of rel_param must lie in [0, 1]",
     fixed = TRUE)
   expect_error(
-    check_rel_susceptibility(1.1),
-    "All values of 'rel_susceptibility' must lie in [0, 1]",
+    check_rel_param(1.1, "rel_param"),
+    "All values of rel_param must lie in [0, 1]",
     fixed = TRUE)
   expect_error(
-    check_rel_susceptibility(c(1, 1.8)),
-    "All values of 'rel_susceptibility' must lie in [0, 1]",
+    check_rel_param(c(1, 1.8), "rel_param"),
+    "All values of rel_param must lie in [0, 1]",
     fixed = TRUE)
   expect_error(
-    check_rel_susceptibility(c(0.9, 0.8)),
-    "First value of 'rel_susceptibility' must be 1")
+    check_rel_param(t(c(0.9, 0.8)), "rel_param"),
+    "First value of rel_param must be 1")
 })
 
 
-test_that("check_rel_susceptibility allows sensible inputs", {
+test_that("check_rel_param allows sensible inputs", {
   expect_silent(
-    check_rel_susceptibility(c(1, 0.5, 0.7)))
+    check_rel_param(t(c(1, 0.5, 0.7)), "rel_param"))
   expect_silent(
-    check_rel_susceptibility(c(1, 0.7, 0.5)))
+    check_rel_param(t(c(1, 0.7, 0.5)), "rel_param"))
   expect_silent(
-    check_rel_susceptibility(1))
+    check_rel_param(t(1), "rel_param"))
   expect_silent(
-    check_rel_susceptibility(c(1, 1)))
+    check_rel_param(t(c(1, 1)), "rel_param"))
   expect_silent(
-    check_rel_susceptibility(c(1, 0)))
+    check_rel_param(t(c(1, 0)), "rel_param"))
   expect_silent(
-    check_rel_susceptibility(c(1, 0, 1)))
+    check_rel_param(t(c(1, 0, 1)), "rel_param"))
 })
 
-test_that("build_rel_susceptibility rejects objects of the wrong dimension", {
-  expect_error(
-    build_rel_susceptibility(
-    rel_susceptibility = matrix(c(1, 0.5, 1, 0.7), nrow = 2, byrow = TRUE)),
-    "'rel_susceptibility' should have as many rows as age groups")
-})
 
-test_that("build_vaccination_rate rejects insensible inputs", {
+test_that("build_rel_param rejects wrong dimension or out of bound inputs", {
   expect_error(
-    build_vaccination_rate(vaccination_rate = c(1, 1)),
-    "'vaccination_rate' should have as many elements as age groups")
+    build_rel_param(
+    rel_param = matrix(c(1, 0.5, 1, 0.7), nrow = 2, byrow = TRUE),
+    n_vacc_classes = 2, "rel_param"),
+    "rel_param should have as many rows as age groups")
   expect_error(
-    build_vaccination_rate(vaccination_rate = c(rep(1, 18), -1)),
-    "'vaccination_rate' must have only non-negative values")
+    build_rel_param(10, n_vacc_classes = 1, "rel_param"),
+    "All values of rel_param must lie in [0, 1]", fixed = TRUE)
 })
 
-test_that("build_vaccination_rate allows sensible inputs & works as expected", {
-  expect_silent(
-    build_vaccination_rate(vaccination_rate = rep(1, 19)))
-  expect_silent(
-    build_vaccination_rate(vaccination_rate = 1))
-  expect_equal(build_vaccination_rate(vaccination_rate = 1), rep(1, 19))
+
+test_that("build_rel_param works as ecpected", {
```

Comment:
just in case it bothers you

```suggestion
test_that("build_rel_param works as expected", {
```

Reply:
done